src/jit/RegisterAlloc.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2023-present Samsung Electronics Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Walrus.h"
+
+#include "jit/Compiler.h"
+#include "jit/SljitLir.h"
+
+#include <set>
+
+namespace Walrus {
+
+void JITCompiler::allocateRegisters()
+{
+    // Dummy register allocator.
+    for (InstructionListItem* item = m_first; item != nullptr; item = item->next()) {
+        if (item->isLabel()) {
+            continue;
+        }
+
+        Instruction* instr = item->asInstruction();
+        Operand* operand = instr->operands();
+        const uint8_t* list = instr->getOperandDescriptor().list;
+        uint32_t paramCount = instr->paramCount();
+        uint32_t resultCount = instr->resultCount();
```

Comment:
I'm a little bit confused with the naming of `param` and `result` here.
`param` represents source operands and `result` for dst operands?

Reply:
One more thing)
It seems like that instructions can have up to 2 param (source?) operands, which means that instruction cannot have more than 2 source operands?

Reply:
This is a WebAssembly style naming. Param is source, result is destination (this can be changed). Source operand limit is 3, for example `select` or `V128BitSelect` has 3 input operands.

Reply:
I see.
We may need to discuss about the naming of these variables later :)

Reply:
No problem. We can land a patch anytime which changes naming. Let us know your preference.